juju/juju
# refactor: remove common model watcher tests

This commit removes the common model watcher test suite from the provisioner facade. This allows us to remove the defunct test suite from common and another usage of state based model config.

The provisioner has been swapped over to using the new service backed common model config methods. Because of this we have removed the one test we had in favour of the tests on the common package.

Also performed several clean up tasks to satisfy some old todo's in the package.

## Checklist


## QA steps

Run the unit tests inside the package and confirm everything is happy. I removed one test as we have test coverage on this from the common package and there is no need to double up.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-6399
